### PR TITLE
evpn: fix nil dereference in shouldDeleteNeighbors

### PR DESCRIPTION
**Pull request by Ultimate-etamitlU**
## Summary
- Fix nil pointer dereference in `shouldDeleteNeighbors` when `SourcePod`
  is nil after completed KubeVirt live migration
- Add defensive nil guard in `shouldEnsureNeighbors` for `TargetPod`
- Add unit tests for post-migration reconciliation with nil SourcePod

## Root Cause
`DiscoverLiveMigrationStatus` returns `LiveMigrationStatus{SourcePod: nil,
TargetPod: pod, State: TargetDomainReady}` when only the target pod
survives post-migration. The EVPN controller dereferences `SourcePod`
without a nil check, causing SIGSEGV that crash-loops `ovnkube-node` and
stalls the `network` ClusterOperator during upgrades.

Two paths in `DiscoverLiveMigrationStatus` produce this state:
1. Single VM pod with `migration-target-ready-timestamp` annotation (line 580)
2. Multiple VM pods but only target is living with the annotation (line 615)

## How to verify it
Existing KubeVirt+EVPN live migration E2E tests (`test/e2e/kubevirt.go`,
"should keep ip" DescribeTable) exercise the full migration lifecycle
including post-migration reconciliation. The crash path is hit naturally
when the source pod is garbage collected after migration completes.

A dedicated E2E test for nil SourcePod is not feasible without mocking
KubeVirt's pod lifecycle — the unit tests directly cover both nil paths
returned by DiscoverLiveMigrationStatus.

No documentation update is needed — this is a defensive nil guard fix,
not a behavior or API change. The EVPN/live-migration documentation
does not describe internal reconciliation implementation details.

## Test plan
- [x] New unit tests: post-migration nil SourcePod on local and remote node
- [x] All 48 existing EVPN controller unit tests pass
- [ ] E2E: existing KubeVirt+EVPN live migration tests cover this path

## Release Notes
Fix ovnkube-node crash loop caused by nil pointer dereference in EVPN
controller during post-migration reconciliation when the source pod
no longer exists.

Assisted-by: Claude Opus 4.6

### CODE REVIEW

File: go-controller/pkg/node/controllers/evpn/evpn_pod_controller_test.go [hunk 1]
```diff
@@ -447,6 +447,35 @@ var _ = Describe("EVPN pod controller", func() {
...
+				// shouldDeleteNeighbors must handle nil SourcePod without panic.
+				targetPod := newVirtLauncherPod("virt-launcher-target", nodeName, 0, true)
+				ctrl.podLister = newFakePodLister(targetPod)
+
+				nlMock.On("NeighSet", mock.Anything).Return(nil)
+
+				Expect(ctrl.reconcilePod("test-ns/virt-launcher-target")).To(
+					Succeed(), "target-only post-migration reconciliation should succeed",
+				)
+			})
```
[the remark this answers is not in the document]
Comment (Ultimate-etamitlU) [reply go-controller/pkg/node/controllers/evpn/evpn_pod_controller_test.go:462]: The TargetPod == nil guard in shouldEnsureNeighbors is purely defensive — there is no code path in DiscoverLiveMigrationStatus that returns LiveMigrationStatus with TargetPod == nil and State == TargetDomainReady. All three paths that set TargetDomainReady also set TargetPod to a non-nil value:

 - Line 580: TargetPod: vmPods[0]
 - Line 615: TargetPod: targetPod
 - Line 634: status.TargetPod = livingPods[1]

Testing this guard via reconcilePod is not feasible without bypassing DiscoverLiveMigrationStatus entirely, which would test a scenario that cannot occur in practice. The guard exists as a safety net against future refactors, not against a known bug path.

**Review by trozet — APPROVED** [review]
